### ISSUE TITLE
GPT-2 and MPT produce empty output (0 tokens) from GGUF models

### ISSUE THREAD

**Issue by cagyirey** [EricLBuehler/mistral.rs#1953]
## Problem

GPT-2 and MPT GGUF models load and run inference without errors, but produce 0 output tokens. The models correctly load weights, initialize caches, and accept prompts, but the generated text is empty.

## Observed Behavior

Both models complete the prompt step and return 0 completion tokens:

- **GPT-2**: `QuantFactory/gpt2-GGUF` / `gpt2.Q4_K_M.gguf` — empty output
- **MPT**: `LoneStriker/mpt-7b-chat-GGUF` / `mpt-7b-chat.Q4_K_M.gguf` — empty output

The golden regression output for both is an empty string.

## Possible Causes

1. **EOS at first token**: The model may predict EOS as the first completion token, causing generation to stop immediately.
2. **Tokenizer decode issue**: The tokenizer may not correctly decode the generated token IDs.
3. **Logits corruption**: The LM head may be producing degenerate logits (all zeros, NaN, etc.).

## Reproduction

```bash
cd /path/to/geodesic
cargo test --features cuda --test regression -- gpt2::regenerate_golden --ignored --nocapture
cargo test --features cuda --test regression -- mpt::regenerate_golden --ignored --nocapture
```

## Metadata

- GPT-2 arch: `gpt2` (GGUF), PostNormTransformerBlock pattern
- MPT arch: `mpt` (GGUF), ALiBi attention (no RoPE)
- Both have tied embeddings (output = token_embd)

Labels: bug, model-arch
